### DISCUSSION TITLE
vitest and serveStatic result in Error: Cannot find package '__STATIC_CONTENT_MANIFEST'

### DISCUSSION THREAD

**Discussion by jbasdf** [honojs/hono#1172] [Q&A]
I'm using vitest to test my hono application that is built to run on Cloudflare. I just added serveStatic to be able to serve static assets and now I'm getting the following error when I run my tests:

Error: Cannot find package '__STATIC_CONTENT_MANIFEST' imported from /Users/{username}/projects/atomic-lti-worker/node_modules/hono/dist/adapter/cloudflare-workers/server-static-module.js

I've added the following to wrangler.toml:
[site]
bucket = "./assets"

The application runs fine it's just the tests that are failing. I tried doing this in vite.config.js but I can't seem to get vite to alias "__STATIC_CONTENT_MANIFEST".

import { defineConfig } from 'vitest/config';
import path from 'path';

export default defineConfig({
  test: {
    environment: 'miniflare',
    alias: {
      __STATIC_CONTENT_MANIFEST: path.resolve(__dirname, './test/manifest.ts'),
    },
  },
});

These are the lines where I use serveStatic:
app.use('/*', serveStatic({ root: './' }));
app.use('/favicon.ico', serveStatic({ path: './favicon.ico' }));
app.use('/robots.txt', serveStatic({ path: './robots.txt' }));

If I comment those out everything runs fine. 

Any tips on how to get this working?

Any tips for getting this working?

**Comment by yusukebe** [comment]
There are some issues when testing it with Vitest and a Miniflare environment. Thus, it's recommended to use [unstable_dev](https://developers.cloudflare.com/workers/wrangler/api/#unstable_dev) in `wrangler`:

```ts
import { expect, describe, beforeAll, afterAll, it } from 'vitest'
import { unstable_dev } from 'wrangler'

describe('Worker', () => {
  let worker

  beforeAll(async () => {
    worker = await unstable_dev('src/index.ts')
  })

  afterAll(async () => {
    await worker.stop()
  })

  it('should return Hello World', async () => {
    const resp = await worker.fetch()
    if (resp) {
      const text = await resp.text()
      expect(text).toMatchInlineSnapshot(`"Hello World!"`)
    }
  })
})
```

**Reply by jbasdf** [reply]
I have quite a few tests written already. Would it be possible to somehow stub out __STATIC_CONTENT_MANIFEST? For example, it looks like there is a way to do it using Jest - https://github.com/cloudflare/miniflare/issues/233#issuecomment-1159444291

**Reply by jbasdf** [reply]
Cloudflare just announced that they are working on merging pages and workers. I solved the problem by switching to pages and removing "serveStatic" then I have Cloudflare serve the static assets.
